src/EA.Iws.DataAccess/Repositories/NotificationAuditRepository.cs

```diff
@@ -0,0 +1,30 @@
+﻿namespace EA.Iws.DataAccess.Repositories
+{
+    using Domain.NotificationApplication;
+    using System;
+    using System.Collections.Generic;
+    using System.Data.Entity;
+    using System.Linq;
+    using System.Threading.Tasks;
+
+    public class NotificationAuditRepository : INotificationAuditRepository
+    {
+        private readonly IwsContext context;
+
+        public NotificationAuditRepository(IwsContext context)
+        {
+            this.context = context;
+        }
+
+        public async Task AddNotificationAudit(Audit notificationAudit)
+        {
+            this.context.NotificationAudit.Add(notificationAudit);
```

Comment:
Can you please add the access check here? You can inject the `INotificationApplicationAuthorization` interface and calliing `EnsureAccessAsync`.